src/device-api/v2.ts

```diff
@@ -85,286 +74,254 @@ export function createV2Api(router: Router, applications: ApplicationManager) {
 							res.status(200).send('OK');
 						});
 				})
-				.catch(err => {
-					res.status(503).send(messageFromError(err));
-				});
+				.catch(next);
 		});
 	};
 
 	router.post(
 		'/v2/applications/:appId/purge',
-		(req: Request, res: Response) => {
+		(req: Request, res: Response, next: NextFunction) => {
 			const { force } = req.body;
 			const { appId } = req.params;
 
 			return doPurge(applications, appId, force)
 				.then(() => {
 					res.status(200).send('OK');
 				})
-				.catch(err => {
-					let message;
-					if (err != null) {
-						message = err.message;
-						if (message == null) {
-							message = err;
-						}
-					} else {
-						message = 'Unknown error';
-					}
-					res.status(503).send(message);
-				});
+				.catch(next);
 		},
 	);
 
 	router.post(
 		'/v2/applications/:appId/restart-service',
-		(req: Request, res: Response) => {
-			return handleServiceAction(req, res, 'restart');
+		(req: Request, res: Response, next: NextFunction) => {
+			return handleServiceAction(req, res, next, 'restart');
```

Comment:
Same for these, it'd be nice to use a partial to reduce duplication

Reply:
Thanks, good point.

Reply:
Addressed. PTAL again.